.github/workflows/os-osd-deployment.yml

```diff
@@ -0,0 +1,131 @@
+# This workflow was used for deploying the latest version of OpenSearch and OpenSearch Dashboards to AWS EKS (Elastic Kubernetes Service) cluster.
+name: Deploy OpenSearch and OpenSearch Dashboards 
+
+on:
+  push:
+    branches: [ main ]
+  # adds workflow_dispatch for manually running a workflow
+  workflow_dispatch:
+
+jobs:
+
+  Pre-Deployment:
+    runs-on: ubuntu-latest
+
+    outputs:
+      config_change_dev: ${{steps.filter.outputs.dev}}
+      config_change_prod: ${{steps.filter.outputs.prod}}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # creats filters for tracking dev and prod changes
+          filters: |
+            dev:
+              - 'config/playground/helm/dev/**'
+            prod:
+              - 'config/playground/helm/prod/**'
+
+  OS-OSD-Dev-Deployment:
+    if: ${{ needs.Pre-Deployment.outputs.config_change_dev == 'true'}}
```

Comment:
I saw this has been used in many places, maybe consider making it an env variable if possible?

Reply:
We need to pass the environment variable to different jobs, not the same job. If we pass it in the same jobs, we can create a environment for it. the way that we pass the value to different jobs is leveraging outputs from job. the reference is here https://docs.github.com/en/actions/learn-github-actions/environment-variables#passing-values-between-steps-and-jobs-in-a-workflow

Reply:
I see, that makes sense.